src/materialized/tests/sql.rs

```diff
@@ -345,6 +345,27 @@ fn test_tail_fetch_timeout() -> Result<(), Box<dyn Error>> {
     let rows = client.query("FETCH c WITH (TIMEOUT = '0s')", &[])?;
     assert_eq!(rows.len(), 0);
 
+    // Make a third cursor. Fetch should return immediately if there are enough
+    // rows, even with a really long timeout.
+    //
+    // Regression test for #6307
+    client.batch_execute(
+        "CLOSE c;
+        DECLARE c CURSOR FOR TAIL t",
+    )?;
+    let before = Instant::now();
+    // NB: This timeout is chosen such that the test will timeout if the bad
+    // behavior occurs.
+    let rows = client.query("FETCH 3 c WITH (TIMEOUT = '1h')", &[])?;
+    let duration = before.elapsed();
+    if rows.len() != 0 {
+        assert_eq!(rows.len(), expected.len());
+        assert!(duration < Duration::from_secs(1));
```

Comment:
I wonder if this could be flakey if lots of tests are run in parallel or something? all of the surrounding tests seem to check that we waited at least as long as the timeout, and didn't exceed that by more than a factor of 10. In contrast, this seems like a much tighter assertion on the response latency.

Not sure that its bad to have that, just wanted to flag that it seemed different than the surrounding tests

Reply:
Yeah, if returning immediately available results takes > 1s then we have some other problems. OTOH that's probably not worth giving some future reader pause about why it's different than the surrounding ones. Fixed